### ISSUE TITLE
Fetch RSS fromfacebook page

### ISSUE THREAD

**Issue by JeroenED** [friendica/friendica#5492]
### Friendica version you encountered the problem

see `example.com/friendica` on your Friendica node for the version information.

**Comment by tobiasd** [comment]
If they provide an Atom/RSS feed you should be albe to follow them now simply by adding that feed URL to your contacts.

**Comment by JeroenED** [comment]
Yes, but facebook doesn't provide rss ;)

This link does create a rss-feed, however it's paying. If we can do that in out of our code it would be more convenient.

**Comment by annando** [comment]
We would possibly need some HTML crawler for that - if someone wants to do this, we can add it as an addon. But personally I admit that I do see other priorities for my future work.

**Comment by MrPetovan** [comment]
Do you want to fetch public posts from a Facebook profile or do you want to provide an RSS feed out of a Facebook profile?

**Comment by JeroenED** [comment]
Just being able to follow public posts of a facebook page. Guess I mentioned rss because I think you can start working from that.

**Comment by MrPetovan** [comment]
Of course, but it already is a Friendica addon, so there's no need for FB -> RSS transcription before we actually ingest the post.

**Comment by JeroenED** [comment]
Guess I wasn't really clear in my feature request, So I'll tell the whole background story behind it.

Since a few days I'm really checking into the RSS feature friendica already has and I thought that it would be nice to have certain pages of facebook I used to follow on my feed. While checking into this I didn't find a solution provided by facebook themselves. (Guess they don't want visitors they can't track when watching pornhub). Asking some people I know if they know this and seemed that this feature was killed several years ago. This day, you can only do this via 3rd party sites like fetchrss.

While thinking about this, I thought should be much easier to have it directly via friendica without any in between conversion and as some sort of contact.
- subscription goes via the url (eg. https://www.facebook.com/whatever) like subscribing to a rss-feed
- New posts of a fb-page should be marked as "Facebook" (instead of Friendica, RSS/Atom, Diaspora*, etc)
- Nice to have for me, but probably required: likes and comments also fetched to friendica. But I won't enable this myself because I commonly hate most comment and flamebait likes.
- Nice to have: ability to like and comment on friendica using a a connector the same way the twitter connector works (Admin adding the keys)

I do believe to do most of this legally (in not being blocked over time) will require a app key which I prefer not to add on my node.
A possible workaround could be (I think) to check if page also has twitter and fetch a [twitrss](https://www.twitrss.me/) (being done for most of my pages ;) )

**Comment by annando** [comment]
Problem with Facebook is that they don't grant the needed permissions for their API to enable such functionality.

**Comment by JeroenED** [comment]
In other words, facebook probably killed a lot of apps when rearranging their permissions after analytica?

While I do believe things can be arranged for organizations, I think this request can be shelved. Since we rely on not 1 global token but a token per node this will be far from a junior job.

**Comment by MrPetovan** [comment]
The move to close the public API off came way before the Cambridge Analytica story.

**Comment by MrPetovan** [comment]
We do rely on a token per node for the Twitter connector, but this is not the biggest hurdle regarding Facebook API. @annando told me time and again what the problem was (we had a Facebook connector), but I still don't remember it.

**Comment by annando** [comment]
You have to apply for the permissions to be granted. Means: You have to explain your application, provide screenshots, credentials, so that they can check on their own, etc.

Then they only will permit the use, when it "enhances the Facebook experience". Also they won't provide permissions to query page information from pages where you aren't the administrator. All API calls that could be used to query the timeline of an user are forbidden.

**Comment by hoergen** [comment]
There is the label "need more info" still on this issue. Is research still going on?

**Comment by annando** [comment]
Not from my side.

**Comment by JeroenED** [comment]
Have no idea whose side? But can be closed if it was up to me!
